src/K9OCRS/Controllers/AccountController.cs

```diff
@@ -254,11 +229,112 @@ public async Task<IActionResult> ChangePassword(PasswordReset passwordReset)
+                tr.Commit();
+            }));
+
+            await Task.WhenAll(tasks);
+            return Ok();
+        }
+
+        [HttpPost("changeinfoadmin")]
+        public async Task<IActionResult> ChangeInfoAdmin([FromBody] User newInfo)
+        {
+            User user = await connectionOwner.Use(conn =>
+            {
+                return dbOwner.Users.GetByID(conn, newInfo.ID);
+            });
+            user.Email = newInfo.Email;
+            user.FirstName = newInfo.FirstName;
+            user.LastName = newInfo.LastName;
+            user.UserRoleID = newInfo.UserRoleID;
+            user.ProfilePictureFilename = newInfo.ProfilePictureFilename;
+
+            var tasks = new List<Task>();
+            tasks.Add(connectionOwner.UseTransaction(async (conn, tr) =>
+            {
+                await dbOwner.Users.Update(conn, tr, user);
+                tr.Commit();
+            }));
+
+            await Task.WhenAll(tasks);
+            return Ok();
+        }
+
+        [HttpPost("createuser")] 
+        public async Task<IActionResult> CreateUser([FromBody] CreateUser accountInfo)
+        {
+            if(await ValidateEmailPassword(accountInfo.Email, accountInfo.Password))
+            {
+                var result = await connectionOwner.Use(conn =>
+                {
+                    User user = new();
+                    user.FirstName = accountInfo.First;
+                    user.LastName = accountInfo.Last;
+                    user.Email = accountInfo.Email;
+                    user.Password = GetHashedPassword(accountInfo.Password);
+                    user.UserRoleID = accountInfo.Role;
+                    return dbOwner.Users.Add(conn, user);
+                });
+                return Ok("Account added");
+            }
+            return StatusCode(400, "Failed to create user");
+        }
+
+        [HttpPost("getuser")]
+        public async Task<IActionResult> GetUser([FromBody] int id)
+        {
+            User user = await connectionOwner.Use(conn =>
+            {
+                return dbOwner.Users.GetByID(conn, id);
+            });
+            UserResult userResult = new UserResult(user);
 
+            return Ok(userResult);
+        }
+
+        [HttpPost("queryusers")]
+        [ProducesResponseType(typeof(IEnumerable<UserResult>), 200)]
+        public async Task<IActionResult> QueryUsers([FromBody] int role)
+        {
+            IEnumerable<User> users;  
+            if(role == 0)
+            {
+                users = await connectionOwner.Use(conn =>
+                {
+                    return dbOwner.Users.GetAll(conn);
+                });
+            }
+            else
+            {
+                users = await connectionOwner.Use(conn =>
+                {
+                    return dbOwner.Users.QueryUsersByRole(conn, role);
+                });
+            }
+            
+            return Ok(users);
```

Comment:
You changed the response type that will be displayed on swagger but didn't actually convert the result from `IEnumerable<User>` to `IEnumerable<UserResult>` before returning it so the front-end is still getting the password and no profile picture url.

This adds the conversion
```suggestion
            var userResults = users.Select(u => new UserResult(u, serviceConstants.storageBasePath));
            
            return Ok(userResults);
```